File: Week_01/G20200343040030/plus-one.js
/**
 * 时间复杂度O(n), 空间复杂度O(1)
 * @param {number[]} digits
 * @return {number[]}
 */
var plusOne = function(digits) {
  for (let i = digits.length - 1; i >= 0; i--) {
    digits[i]++;
    digits[i] %= 10;
    if (digits[i] != 0) return digits;
  }

  return [1, ...digits];
};

File: Week_预习/G20200343040338/plus-one.py
class Solution:
    def plusOne(self, digits: List[int]) -> List[int]:
        num = self.to_num(digits)
        return self.to_digits(num + 1)

    def to_num(self, digits):
        num = 0

        for digit in digits:
            num *= 10
            num += digit

        return num

    def to_digits(self, num):
        return [int(digit) for digit in str(num)]
